egui/src/widgets/plot/items.rs

```diff
@@ -73,13 +73,317 @@ struct ExplicitGenerator {
+                let vertical = tf(vec![vec2(0.0, -1.0), vec2(0.0, 1.0)]);
+                vec![
+                    Shape::line(horizontal, default_stroke),
+                    Shape::line(vertical, default_stroke),
+                ]
+            }
+            MarkerShape::Up => {
+                let offsets = vec![
+                    vec2(0.0, -1.0),
+                    vec2(-0.5 * sqrt_3, 0.5),
+                    vec2(0.5 * sqrt_3, 0.5),
+                ];
+                let points = tf(offsets);
+                vec![Shape::Path {
+                    points,
+                    closed: true,
+                    fill,
+                    stroke,
+                }]
+            }
+            MarkerShape::Down => {
+                let offsets = vec![
+                    vec2(0.0, 1.0),
+                    vec2(-0.5 * sqrt_3, -0.5),
+                    vec2(0.5 * sqrt_3, -0.5),
+                ];
+                let points = tf(offsets);
+                vec![Shape::Path {
+                    points,
+                    closed: true,
+                    fill,
+                    stroke,
+                }]
+            }
+            MarkerShape::Left => {
+                let offsets = vec![
+                    vec2(-1.0, 0.0),
+                    vec2(0.5, -0.5 * sqrt_3),
+                    vec2(0.5, 0.5 * sqrt_3),
+                ];
+                let points = tf(offsets);
+                vec![Shape::Path {
+                    points,
+                    closed: true,
+                    fill,
+                    stroke,
+                }]
+            }
+            MarkerShape::Right => {
+                let offsets = vec![
+                    vec2(1.0, 0.0),
+                    vec2(-0.5, -0.5 * sqrt_3),
+                    vec2(-0.5, 0.5 * sqrt_3),
+                ];
+                let points = tf(offsets);
+                vec![Shape::Path {
+                    points,
+                    closed: true,
+                    fill,
+                    stroke,
+                }]
+            }
+            MarkerShape::Asterisk => {
+                let vertical = tf(vec![vec2(0.0, -1.0), vec2(0.0, 1.0)]);
+                let diagonal1 = tf(vec![vec2(-frac_sqrt_3_2, 0.5), vec2(frac_sqrt_3_2, -0.5)]);
+                let diagonal2 = tf(vec![vec2(-frac_sqrt_3_2, -0.5), vec2(frac_sqrt_3_2, 0.5)]);
+                vec![
+                    Shape::line(vertical, default_stroke),
+                    Shape::line(diagonal1, default_stroke),
+                    Shape::line(diagonal2, default_stroke),
+                ]
+            }
+        }
+    }
+}
+
 /// A series of values forming a path.
 pub struct Curve {
     pub(crate) values: Vec<Value>,
     generator: Option<ExplicitGenerator>,
     pub(crate) bounds: Bounds,
-    pub(crate) stroke: Stroke,
+    pub(crate) marker: Option<Marker>,
+    pub(crate) color: Option<Color32>,
+    pub(crate) width: f32,
```

Comment:
Why replace `Stroke` with color + widht?

Reply:
ae00424 allowed me to go back to `Stroke`